### PR TITLE
feat: improve logo processing

### PR DESCRIPTION
**Pull request by raphael0202**
- fix: filter logos that are almost exactly the same
- feat: add `data->bounding_box` field in logo derived insights
- feat: improve `generate_prediction` function

### CODE REVIEW

File: robotoff/logos.py [hunk 1]
```diff
@@ -327,10 +331,20 @@ def predict_logo_predictions(
...
-    if logo_type not in LOGO_TYPE_MAPPING:
+    """Generate a Prediction from a logo.
+
+    The Prediction may either be created after the annotation of the logo by
+    a human (in which case `automatic_processing` is True), or by infering the
+    logo value from nearest neighbor labels (in which case
+    `automatic_processing` is False).
+
+    Currently, only brand and label logo types are supported: None is returned
+    if the logo type is different, or if the logo_value is None.
```
Comment (GabrielBeFr) [inline robotoff/logos.py:404]: This is true regarding automatic processing but not regarding human annotations isn't it ? I thought we could annotate also packager codes for example.

**Review by GabrielBeFr — APPROVED** [review]

File: robotoff/logos.py (hunk 1 above)
Comment (raphael0202) [reply robotoff/logos.py:404]: Yes we can annotate logos of other type than label and brand, but only for these two types a prediction and insight is generated.

File: robotoff/logos.py [hunk 2]
```diff
@@ -23,6 +24,59 @@
...
+    """Select logos that don't intersect with each other
+    (IoU < `iou_threshold`) and that have a confidence score above
+    `score_threshold`.
+
+    Return a list of (original_idx, logo) tuples.
+    """
+    filtered = []
+    skip_indexes = set()
+    for i in range(len(logos)):
+        if i not in skip_indexes:
```
Comment (GabrielBeFr) [inline robotoff/logos.py:63]: `logo = logos[i]`

File: robotoff/logos.py [hunk 3]
```diff
@@ -23,6 +24,59 @@
...
+            for j in range(i + 1, len(logos)):
+                if (
+                    compute_iou(logos[i]["bounding_box"], logos[j]["bounding_box"])
```
Comment (GabrielBeFr) [inline robotoff/logos.py:66]: compute_iou(logo["bounding_box"], logos[j]["bounding_box"])

File: robotoff/logos.py [hunk 4]
```diff
@@ -23,6 +24,59 @@
...
+                    >= iou_threshold
+                ):
+                    # logos are sorted by descending confidence score, so we ignore
+                    # j logo (logo with lower confidence score)
+                    skip_indexes.add(j)
+
+        logo = logos[i]
```
Comment (GabrielBeFr) [inline robotoff/logos.py:73]: delete this line

**Review by GabrielBeFr — COMMENTED** [review]
A very small comment regarding code.

File: robotoff/logos.py [hunk 5]
```diff
@@ -23,6 +24,59 @@
...
+    (IoU < `iou_threshold`) and that have a confidence score above
+    `score_threshold`.
+
+    Return a list of (original_idx, logo) tuples.
+    """
+    filtered = []
+    skip_indexes = set()
+    for i in range(len(logos)):
+        if i not in skip_indexes:
+            for j in range(i + 1, len(logos)):
```
Comment (GabrielBeFr) [inline robotoff/logos.py:64]: logo = logos[i]

File: robotoff/logos.py [hunk 6]
```diff
@@ -23,6 +24,59 @@
...
+                if (
+                    compute_iou(logos[i]["bounding_box"], logos[j]["bounding_box"])
+                    >= iou_threshold
```
Comment (GabrielBeFr) [inline robotoff/logos.py:67]: compute_iou(logo["bounding_box"], logos[j]["bounding_box"])

File: robotoff/logos.py [hunk 7]
```diff
@@ -23,6 +24,59 @@
...
+                ):
+                    # logos are sorted by descending confidence score, so we ignore
+                    # j logo (logo with lower confidence score)
+                    skip_indexes.add(j)
+
+        logo = logos[i]
+        if logo["score"] >= score_threshold:
```
Comment (GabrielBeFr) [inline robotoff/logos.py:74]: We can thus delete this line

**Review by GabrielBeFr — COMMENTED** [review]
Very small details regarding code

File: robotoff/logos.py (hunk 7 above)
Comment (raphael0202) [reply robotoff/logos.py:74]: why?

**Pull request merged**